### ISSUE TITLE
SDK build error with pulpissimo-riscy/rtl configs

### ISSUE THREAD

**Issue by holysml** [pulp-platform/pulp-sdk#5]
Hi, Pulp SDK team :

Here is error message from my sdk building, any idea/clue?

sdk:pulp-rt:build (user_config_file=/net/cola/data/sawu/proj/pulp-sdk_git/configs/json/pulpissimo-riscy.json): make build install
make[1]: Entering directory `/net/cola/data/sawu/proj/pulp-sdk_git/runtime/pulp-rt'
/net/cola/data/sawu/proj/ri5cy_xcom/pulp_bare/bin/riscv32-unknown-elf-gcc  -march=rv32imcxpulpv2 -D__riscv__ -Os -g -fno-jump-tables -fno-tree-loop-distribute-patterns -Werror  -Wextra -Wall -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wundef -fdata-sections -ffunction-sections -I/net/cola/data/sawu/proj/pulp-sdk_git/pkg/sdk/dev/install/include/io -I/net/cola/data/sawu/proj/pulp-sdk_git/pkg/sdk/dev/install/include -include /net/cola/data/sawu/proj/pulp-sdk_git/runtime/pulp-rt/build/user_config_file..net.cola.data.sawu.proj.pulp-sdk_git.configs.json.pulpissimo-riscy.json/fc_config.h    -MMD -MP -c kernel/init.c -o /net/cola/data/sawu/proj/pulp-sdk_git/runtime/pulp-rt/build/user_config_file..net.cola.data.sawu.proj.pulp-sdk_git.configs.json.pulpissimo-riscy.json/rt/fc/kernel/init.o
**cc1: error: -march=rv32imcxpulpv2: unsupported ISA substring 'xpulpv2'**
make[1]: *** [/net/cola/data/sawu/proj/pulp-sdk_git/runtime/pulp-rt/build/user_config_file..net.cola.data.sawu.proj.pulp-sdk_git.configs.json.pulpissimo-riscy.json/rt/fc/kernel/init.o] Error 1

I did trial to go back rebuilding cross toolchain with :
./configure --prefix=/xxx/ --with-arch=**rv32imcxpulpv2** --with-cmodel=medlow --enable-multilib 

However, there is error msg
cp doc/gcc.1 doc/g++.1
/net/cola/data/sawu/proj/pulp-riscv-gnu-toolchain/build-gcc-newlib-stage1/./gcc/xgcc -B/net/cola/data/sawu/proj/pulp-riscv-gnu-toolchain/build-gcc-newlib-stage1/./gcc/ -nostdinc -x c /dev/null -S -o /dev/null -fself-test=../.././riscv-gcc/gcc/testsuite/selftests
cc1: error: -march=rv32imcxpulpv2: unsupported ISA substring 'xpulpv2'
cc1: internal compiler error: Segmentation fault
0xaf3b2f crash_signal
        ../.././riscv-gcc/gcc/toplev.c:337
0xdb465f tree_class_check(tree_node*, tree_code_class, char const*, int, char const*)
        ../.././riscv-gcc/gcc/tree.h:3199
0xdb465f build_int_cst(tree_node*, long)
        ../.././riscv-gcc/gcc/tree.c:1297
0x83f8c5 test_tree_marking
        ../.././riscv-gcc/gcc/ggc-tests.c:458
0x83f8c5 selftest::ggc_tests_c_tests()
        ../.././riscv-gcc/gcc/ggc-tests.c:485
0x1050543 selftest::run_tests()
        ../.././riscv-gcc/gcc/selftest-run-tests.c:65
0xaf5152 toplev::run_self_tests()
        ../.././riscv-gcc/gcc/toplev.c:2067
Please submit a full bug report,

Any idea/suggestion?

Regards,
Samuel

**Comment by haugoug** [comment]
Looks like you are using either a generic riscv compiler or an old pulp compiler. Make sure you get the latest version from here:
git@github.com:pulp-platform/pulp-riscv-gnu-toolchain.git

**Comment by haugoug** [comment]
Is this fixed ?

**Comment by holysml** [comment]
Not fixed yet, sorry for late reply due to Lunar new year vacation. 

I did install and build pulp-tool chain from https://github.com/pulp-platform/pulp-riscv-gnu-toolchain from beginning.
However, it failed in pulp-sdk build phase.

Build pulp cross toolchain : 
./configure --prefix=/opt/riscv --with-arch=rv32imc --with-cmodel=medlow --enable-multilib

Try to replace as --with-arch=rv32imcxpulpv2 => Failure.
-=-=-=-=-=-=-=-=-=-=-=-=-
Build cross toolchain with --with-arch=rv32imc --with-cmodel=medlow --enable-multilib, toolchain OK.
Then build pulpissimo => Failure, err msg attached in 1st post in this thread. 

It is tricky why pulp-sdk applies -march=rv32imcxpulpv2 but this arch option can't be applied on cross toolchain building?

**Comment by haugoug** [comment]
You should not build the toolchain with --with-arch=rv32imcxpulpv2.
The --march option is just used as the default arch is the -march option is not provided when compiling pulp code.
The toolchain is compiled with the multilib support and thus the runtime will be compiled for all variants.
If the toolchain building went well, you should find for example this library in your toolchain install folder:
./riscv32-unknown-elf/lib/rv32imcxpulpv2/ilp32/libc.a
Do you have it ?

**Comment by holysml** [comment]
Hi, haugoug :

I don't get something under ./riscv32-unknown-elf/lib/rv32imcxpulpv2 ...
![image](https://user-images.githubusercontent.com/17939673/36514253-8b580ac6-17ae-11e8-824d-fe0b17f6e636.png)

I should check toolcahin compilation log, can you help me some clue in the mean time?

**Comment by haugoug** [comment]
The README was still showing the generic riscv URL for the git clone command, did you take the one for Pulp ?
I've just updated the Readme to show the pulp URL

**Comment by holysml** [comment]
Hi

I did use repository https://github.com/pulp-platform/pulp-riscv-gnu-toolchain from beginning; however I did trial with success built process in approach : 
**git clone --recursive https://github.com/pulp-platform/pulp-riscv-gnu-toolchain**

rather than 2nd alternative(I followed 2nd approach from beginning).

And cross compiler did work; proceed to sdk building progress.
Thank your input.

Regards,
Samuel
